src/main/kotlin/no/skatteetaten/aurora/boober/service/HerkimerService.kt

```diff
@@ -0,0 +1,161 @@
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule
+import no.skatteetaten.aurora.boober.ServiceTypes
+import no.skatteetaten.aurora.boober.TargetService
+import no.skatteetaten.aurora.boober.utils.RetryingRestTemplateWrapper
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestTemplate
+import java.time.LocalDateTime
+
+data class HerkimerResponse<T : Any>(
+    val success: Boolean = true,
+    val message: String = "OK",
+    val items: List<T> = emptyList(),
+    val errors: List<ErrorResponse> = emptyList(),
+    val count: Int = items.size + errors.size
+)
+
+data class ErrorResponse(val errorMessage: String)
+data class ResourceClaimPayload(
+    val ownerId: String,
+    val credentials: Any
+)
+
+data class ApplicationDeploymentPayload(
+    val name: String,
+    val environmentName: String,
+    val cluster: String,
+    val businessGroup: String,
+    val applicationName: String
+)
+
+data class ApplicationDeploymentHerkimer(
+    val id: String,
+    val name: String,
+    val environmentName: String,
+    val cluster: String,
+    val businessGroup: String,
+    val applicationName: String,
+    val createdDate: LocalDateTime,
+    val modifiedDate: LocalDateTime,
+    val createdBy: String,
+    val modifiedBy: String
+)
+
+data class ResourceHerkimer(
+    val id: String,
+    val name: String,
+    val kind: ResourceKind,
+    val ownerId: String,
+    val claims: List<ResourceClaimHerkimer>? = null,
+    val createdDate: LocalDateTime,
+    val modifiedDate: LocalDateTime,
+    val createdBy: String,
+    val modifiedBy: String
+)
+
+data class ResourceClaimHerkimer(
+    val id: String,
+    val ownerId: String,
+    val resourceId: Long,
+    val credentials: JsonNode,
+    val createdDate: LocalDateTime,
+    val modifiedDate: LocalDateTime,
+    val createdBy: String,
+    val modifiedBy: String
+)
+
+enum class ResourceKind {
+    MinioPolicy, ManagedPostgresDatabase, ManagedOracleSchema, ExternalSchema
+}
+
+data class ResourcePayload(
+    val name: String,
+    val kind: ResourceKind,
+    val ownerId: String
+)
+
+@Component
+class HerkimerRestTemplateWrapper(@TargetService(ServiceTypes.HERKIMER) restTemplate: RestTemplate, @Value("\${integrations.herkimer.retries:3}")override val retries: Int) :
+    RetryingRestTemplateWrapper(restTemplate = restTemplate, retries = retries)
+
+@Service
+class HerkimerService(
+    val client: HerkimerRestTemplateWrapper,
+    objectMapper: ObjectMapper
+) {
+    private val herkimerObjectmapper = objectMapper.configureHerkimerDefaults()
```

Comment:
Hmm... denne er litt annerledes. Endrer configureHerkimerDefaults på this, eller lager den en ny kopi? Dersom HerkimerService trenger en helt spesiell konfig på ObjectMapper, er det nesten bedre at den bare lager den selv, og ikke får en injectet?

Reply:
Jeg er jammen litt usikker når jeg tenker meg om. Det er nok best å gjøre som du sier og lage en egen. Ikke vits å injecte den

Reply:
Ja, dette er en litt spesiell situasjon, potensielt, dersom deserialisering av herkimer-responser må følge en bestemt config av ObjectMapper.